pageserver/src/tenant/timeline.rs

```diff
@@ -4671,27 +4671,6 @@ impl Timeline {
             }
         }
 
-        // The writer.finish() above already did the fsync of the inodes.
-        // We just need to fsync the directory in which these inodes are linked,
-        // which we know to be the timeline directory.
-        if !image_layers.is_empty() {
-            // We use fatal_err() below because the after writer.finish() returns with success,
-            // the in-memory state of the filesystem already has the layer file in its final place,
-            // and subsequent pageserver code could think it's durable while it really isn't.
-            let timeline_dir = VirtualFile::open(
-                &self
-                    .conf
-                    .timeline_path(&self.tenant_shard_id, &self.timeline_id),
-                ctx,
-            )
-            .await
-            .fatal_err("VirtualFile::open for timeline dir fsync");
-            timeline_dir
-                .sync_all()
-                .await
-                .fatal_err("VirtualFile::sync_all timeline dir");
```

Comment:
Why is this spurious? `ImageLayerWriter::finish` calls into libc `renameat2` which translates into a `rename` syscall. I though fsyncing the directory is still required.

Reply:
Across a crash/restart, only layers that have been uploaded to S3 will be downloaded.  fsync'ing the dentries for layers is an optimization that only provides benefit in the extremely rare case that filesystem metadata hasn't hit disk yet, _but_ the layer+index have already hit S3 (i.e. without the fsync, on restart we would re-download the layer).

Sorry, bit random of me to sneak this change into this PR, it was something I noticed while checking over what this function was doing.

Reply:
Ah yes - makes sense. I don't have a strong opinion on this trade-off.